### PR TITLE
Allow usage of env var for the namespace

### PR DESCRIPTION
**Pull request by Johnmeurt**
I move the validation of namespace into the extension to allow usage of env var. The env var will be resolved on compilation time only and set in cache. Change in runtime will have no effect.

The rules on the namespace are still the same.

### CODE REVIEW

File: Tests/DependencyInjection/ConfigurationTest.php [hunk 1]
```diff
@@ -152,19 +152,38 @@ public function configDataProvider(): array
                     'disable_default_promphp_metrics' => false,
                 ],
             ],
+            [
+                'Namesapce as dsn',
```
Comment (denisvmedia) [inline Tests/DependencyInjection/ConfigurationTest.php:156]:
```suggestion
                'Namespace as dsn',
```

File: Tests/DependencyInjection/ConfigurationTest.php [hunk 2]
```diff
@@ -152,19 +152,38 @@ public function configDataProvider(): array
...
+            ],
         ];
     }
 
     public function invalidConfigDataProvider(): array
     {
         return [
             [
-                'invalid namespace (with dashes)',
+                'Invalid prefix',
```
Comment (denisvmedia) [inline Tests/DependencyInjection/ConfigurationTest.php:178]: Why did you decide to change this?

**Comment by denisvmedia** [comment]
Thank you for your efforts. There is a question I left and a minor typo in your code. I will merge it, as soon as we sort the things out.

File: Tests/DependencyInjection/ConfigurationTest.php (hunk 1 above)
Comment (Johnmeurt) [reply Tests/DependencyInjection/ConfigurationTest.php:156]: Hum... I should remove this test cause it has no sense. the string "%env(...)%" is never injected into the config array.

**Pull request merged**